aten/src/ATen/native/native_functions.yaml

```diff
@@ -3255,6 +3255,40 @@
       SparseCPU: _sparse_sum_backward_cpu
       SparseCUDA: _sparse_sum_backward_cuda
 
+- func: _sparse_softmax.int(Tensor self, int dim, ScalarType? dtype=None) -> Tensor
+  variants: function
+  supports_named_tensor: True
+
+- func: _sparse_softmax.Dimname(Tensor self, Dimname dim, *, ScalarType? dtype=None) -> Tensor
+  variants: function
+  supports_named_tensor: True
+
+- func: _sparse_softmax(Tensor self, int dim, bool half_to_float) -> Tensor
+  use_c10_dispatcher: full
+  dispatch:
+    SparseCPU: softmax_sparse_cpu
+
+- func: _sparse_softmax_backward_data(Tensor grad_output, Tensor output, int dim, Tensor self) -> Tensor
+  dispatch:
+    SparseCPU: softmax_backward_sparse_cpu
```

Comment:
@albanD , could you please tell whether backward for a backward is required here?

Reply:
Given that the backaward implementation cannot be automatically differentiated, you will get an error if you try to differentiate through this.
In general, we don't block a feature on not having a double backward.
If you want it here though, you can define a formula for it as `softmax_backward_backward_sparse_cpu` in `tools/autograd/template/Functions.cpp` that only uses high level pytorch ops so that it will be infinitely differentiable.

Reply:
With the sparse tensor limited support for high level pytorch operations, I suggest postponing the double backward for sparse softmax.